### PR TITLE
Magnetic field of dipoles in Cartesian coordinates

### PR DESCRIPTION
**Pull request by santisoler**
Add forward modelling functions for the magnetic field of dipoles in Cartesian coordinates: `dipole_magnetic` and `dipole_magnetic_component`. The first one computes the three components of the magnetic field, while the latter computes only one of them. The forward modelling is done using the Choclo functions for dipoles.

**Relevant issues/PRs:**

Part of #406

### CODE REVIEW

File: harmonica/_forward/dipole.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+        ``np.float64``.
+    disable_checks : bool (optional)
+        Flag that controls whether to perform a sanity check on the model.
+        Should be set to ``True`` only when it is certain that the input model
+        is valid and it does not need to be checked.
+        Default to ``False``.
+
+    Returns
+    -------
+    magnetic_field : tuple of array
```
Comment (santisoler) [inline harmonica/_forward/dipole.py:57]:
```suggestion
    magnetic_field : tuple of arrays
```

File: harmonica/_forward/dipole.py [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
```
Comment (santisoler) [inline harmonica/_forward/dipole.py:149]:
```suggestion
    magnetic_field : tuple of arrays
```

File: harmonica/_forward/dipole.py [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+        Tuple containing each component of the magnetic field generated by the
+        dipoles as arrays. The three components are returned in the following
+        order: ``b_e``, ``b_n``, ``b_u``.
```
Comment (santisoler) [inline harmonica/_forward/dipole.py:60]:
```suggestion
        Tuple containing each component of the magnetic field generated by the
        dipoles as arrays. The three components are returned in the following
        order: ``b_e``, ``b_n``, ``b_u``. Magnetic field components are in :math:`nT`.
```

File: doc/user_guide/forward_modelling/dipole.rst [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+   b_u = hm.dipole_magnetic_component(
+      coordinates, dipole, magnetic_moment, component="upward"
+   )
+   print(b_u)
+
+
+Multiple dipoles
+----------------
+
+These two functions also allows us to compute the magnetic fields of multiple
```
Comment (LL-Geo) [inline doc/user_guide/forward_modelling/dipole.rst:59]:
```suggestion
These two functions also allow us to compute the magnetic fields of multiple
```

File: harmonica/_forward/dipole.py [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+    component : str
+        Computed that will be computed. Available options are: ``"easting"``,
+        ``"northing"`` or ``"upward"``.
+    parallel : bool (optional)
+        If True the computations will run in parallel using Numba built-in
+        parallelization. If False, the forward model will run on a single core.
+        Might be useful to disable parallelization if the forward model is run
+        by an already parallelized workflow. Default to True.
+    dtype : data-type (optional)
+        Data type assigned to the resulting gravitational field. Default to
```
Comment (LL-Geo) [inline harmonica/_forward/dipole.py:131]:
```suggestion
        Data type assigned to the resulting magnetic field. Default to
```

**Review by LL-Geo — COMMENTED** [review]
Looks great @santisoler !😁 A small change for the figure and some typos then it will be perfect to merge 😀
image

File: harmonica/_forward/dipole.py [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+        :math:`Am^2`. Each row contains the components of the magnetic moment
+        for each dipole in the following order: ``mag_moment_easting``,
+        ``mag_moment_northing``, ``mag_moment_upward``.
+    parallel : bool (optional)
+        If True the computations will run in parallel using Numba built-in
+        parallelization. If False, the forward model will run on a single core.
+        Might be useful to disable parallelization if the forward model is run
+        by an already parallelized workflow. Default to True.
+    dtype : data-type (optional)
+        Data type assigned to the resulting gravitational field. Default to
```
[the remark this answers is not in the document]
Comment (santisoler) [reply harmonica/_forward/dipole.py:47]: Oops, my bad. Nice catch!

**Pull request merged**